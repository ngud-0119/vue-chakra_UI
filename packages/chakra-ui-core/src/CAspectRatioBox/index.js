import AspectRatioBox from './AspectRatioBox'
export default AspectRatioBox
