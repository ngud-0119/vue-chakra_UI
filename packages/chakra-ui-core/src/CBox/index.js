import CBox from './CBox'
export default CBox
export * from './CBox'
