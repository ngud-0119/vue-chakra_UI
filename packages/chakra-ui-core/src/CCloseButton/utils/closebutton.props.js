export default {
  size: {
    type: String,
    default: 'md',
    validator: value => value.match(/^(sm|md|lg)$/)
  },
  isDisabled: {
    type: Boolean,
    default: false
  },
  ariaLabel: {
    type: String,
    default: 'Close'
  }
}
