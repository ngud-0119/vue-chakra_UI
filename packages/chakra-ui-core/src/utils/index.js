export { default as Logger } from './logger'
export { provideTheme } from './provide-theme'
export { transformAlias as tx, valueToPercent } from './transform'
export { pickProperty as forwardProps, filterBaseStyles, filterPseudo, unwrapValues } from './object'
export * from './color'
export { parsePackIcons } from './icons'
export * from './dom'
export * from './vdom'
export * from './strings'
export * from './generators'
export * from './validators'
export * from './functions'
export * from './components'
export { proxyAliases as pxls } from '../config/props/proxy'
