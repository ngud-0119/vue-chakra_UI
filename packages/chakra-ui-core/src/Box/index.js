
import { css } from 'emotion'
import { background, border, color, borderRadius, flexbox, grid, layout, position, shadow, space, typography, compose } from 'styled-system'
import { baseProps, propsConfig } from '../config/props'
import { forwardProps } from '../utils'

const baseEllipsis = {
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
}

/**
 * @description Truncates text if `truncate` is set to true.
 * @param {Object} props Props
 */
const truncate = props => {
  if (props.truncate) {
    if (!props.lineClamp) {
      return baseEllipsis
    }
  }
}

/**
 * @description Clamps text based on number of lines.
 * @param {Object} props Props
 */
const clamp = props => {
  if (props.lineClamp) {
    return {
      ...baseEllipsis,
      '-webkit-box-orient': 'vertical',
      '-webkit-line-clamp': `${props.lineClamp}`
    }
  }
}

const decorate = props => {
  if (props.textDecoration || props.textDecor) {
    return {
      'text-decoration': `${props.textDecoration || props.textDecor}`
    }
  }
}

const systemProps = compose(
  space,
  layout,
  color,
  background,
  border,
  borderRadius,
  grid,
  position,
  shadow,
  decorate,
  typography,
  flexbox,
  propsConfig,
  truncate,
  clamp
)

const Box = {
  name: 'Box',
  inject: ['$theme'],
  props: {
    as: {
      type: [String, Object],
      default: 'div'
    },
    ...baseProps
  },
  computed: {
    theme () {
      return this.$theme()
    }
  },
  render (h) {
    const { as, ...cleanedStyleProps } = forwardProps(this.$props)
    const boxStylesObject = systemProps({ ...cleanedStyleProps, theme: this.theme })
    const className = css(boxStylesObject)

    return h(as, {
      class: [className]
    }, this.$slots.default)
  }
}

export default Box
