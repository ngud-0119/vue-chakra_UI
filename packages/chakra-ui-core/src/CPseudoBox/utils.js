import map from 'lodash-es/map'
import { forwardProps, filterBaseStyles, filterPseudo as getPseudoStyles, tx } from '../utils'

/**
 * PseudoBox pseudo selectors
 */
export const selectors = {
  _hover: '&:hover',
  _active: '&:active, &[data-active=true]',
  _focus: '&:focus',
  _visited: '&:visited',
  _even: '&:nth-of-type(even)',
  _odd: '&:nth-of-type(odd)',
  _disabled: '&:disabled, &:disabled:focus, &:disabled:hover, &[aria-disabled=true], &[aria-disabled=true]:focus, &[aria-disabled=true]:hover',
  _checked: '&[aria-checked=true]',
  _mixed: '&[aria-checked=mixed]',
  _selected: '&[aria-selected=true]',
  _invalid: '&[aria-invalid=true]',
  _pressed: '&[aria-pressed=true]',
  _readOnly: '&[aria-readonly=true], &[readonly]',
  _first: '&:first-of-type',
  _last: '&:last-of-type',
  _expanded: '&[aria-expanded=true]',
  _grabbed: '&[aria-grabbed=true]',
  _notFirst: '&:not(:first-of-type)',
  _notLast: '&:not(:last-of-type)',
  _groupHover: '[role=group]:hover &',
  _before: '&:before',
  _after: '&:after',
  _focusWithin: '&:focus-within',
  _placeholder: '&::placeholder'
}

/**
 * Filter undefined props and parse pseudo style props to generate css styles object
 * @param {Object} props - Props
 * @returns {Object} Parsed pseudo styles object
 */
export function parsePseudoStyles (props) {
  const pseudoStyles = {}
  const clean = forwardProps(props)
  const pseudoProps = getPseudoStyles(clean)
  const baseProps = filterBaseStyles(clean)
  const result = map(pseudoProps, (value, prop) => ({ prop, value }))
  result.forEach((pair) => {
    if (selectors[pair.prop]) {
      pseudoStyles[selectors[pair.prop]] = tx(pair.value)
    }
  })
  const mergedPseudoBoxStyles = { pseudoStyles, baseProps }
  return mergedPseudoBoxStyles
}
