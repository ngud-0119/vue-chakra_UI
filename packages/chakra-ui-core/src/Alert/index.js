import Box from '../Box'
import Icon from '../Icon'
import { baseProps } from '../config/props'
import { forwardProps } from '../utils'
import useAlertStyle, { useAlertIconStyle } from './alert.styles'

export const statuses = {
  info: { icon: '_info', color: 'blue' },
  warning: { icon: '_warning-2', color: 'orange' },
  success: { icon: '_check-circle', color: 'green' },
  error: { icon: '_warning', color: 'red' }
}

const Alert = {
  name: 'Alert',
  inject: ['$theme', '$colorMode'],
  provide () {
    return {
      _status: this.status,
      _variant: this.variant
    }
  },
  computed: {
    colorMode () {
      return this.$colorMode()
    }
  },
  props: {
    status: {
      type: [String, Array],
      default: 'info'
    },
    variant: {
      type: [String, Array],
      default: 'subtle'
    },
    ...baseProps
  },
  render (h) {
    const alertStyles = useAlertStyle({
      variant: this.variant,
      color: statuses[this.status] && statuses[this.status]['color'],
      colorMode: this.colorMode,
      theme: this.$theme()
    })

    return h(Box, {
      props: {
        fontFamily: 'body',
        ...alertStyles,
        ...forwardProps(this.$props)
      },
      attrs: {
        role: 'alert'
      }
    }, this.$slots.default)
  }
}

const AlertIcon = {
  name: 'AlertIcon',
  inject: ['_status', '_variant', '$colorMode', '$theme'],
  props: {
    size: {
      default: 5
    },
    name: String,
    ...baseProps
  },
  computed: {
    colorMode () {
      return this.$colorMode()
    }
  },
  render (h) {
    const alertIconStyles = useAlertIconStyle({
      variant: this._variant,
      colorMode: this.colorMode,
      color: statuses[this._status] && statuses[this._status]['color']
    })

    return h(Icon, {
      props: {
        mr: this.$props.mr || 3,
        size: this.size,
        name: this.name || (statuses[this._status] && statuses[this._status]['icon']),
        ...alertIconStyles,
        ...forwardProps(this.$props)
      },
      attrs: {
        focusable: false
      }
    })
  }
}

const AlertTitle = {
  name: 'AlertTitle',
  props: {
    ...baseProps
  },
  render (h) {
    return h(Box, {
      props: {
        fontWeight: 'bold',
        lineHeight: 'normal',
        ...forwardProps(this.$props)
      }
    }, this.$slots.default)
  }
}

const AlertDescription = {
  name: 'AlertDescription',
  props: {
    ...baseProps
  },
  render (h) {
    return h(Box, { props: forwardProps(this.$props) }, this.$slots.default)
  }
}

export { Alert, AlertIcon, AlertTitle, AlertDescription }
