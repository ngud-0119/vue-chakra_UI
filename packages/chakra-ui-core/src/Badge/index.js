import Box from '../Box'
import { forwardProps } from '../utils'
import { baseProps } from '../config/props'
import useBadgeStyles from './badge.styles'

export default {
  name: 'Badge',
  inject: ['$theme', '$colorMode'],
  props: {
    variant: {
      type: String,
      default: 'subtle'
    },
    variantColor: {
      type: String,
      default: 'gray'
    },
    ...baseProps
  },
  computed: {
    colorMode () {
      return this.$colorMode()
    }
  },
  render (h) {
    const badgeStyleProps = useBadgeStyles({
      theme: this.$theme(),
      colorMode: this.colorMode,
      color: this.variantColor,
      variant: this.variant
    })

    return h(Box, {
      props: {
        ...forwardProps(this.$props),
        d: 'inline-block',
        textTransform: 'uppercase',
        fontSize: 'xs',
        fontFamily: 'body',
        px: 1,
        rounded: 'sm',
        fontWeight: 'bold',
        whiteSpace: 'nowrap',
        verticalAlign: 'middle',
        ...badgeStyleProps
      }
    }, this.$slots.default)
  }
}
