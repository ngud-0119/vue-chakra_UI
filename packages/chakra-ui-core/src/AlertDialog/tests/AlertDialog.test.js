import { Button, AlertDialog, AlertDialogContent, AlertDialogBody, AlertDialogFooter, AlertDialogOverlay, AlertDialogHeader, Portal } from '../../'
import { render, defaultProviders, userEvent, fireEvent } from '@/tests/test-utils'
import Vue from 'vue'
import { useId, wrapEvent } from '@/packages/chakra-ui-core/src/utils'

// mocks
jest.mock('@/packages/chakra-ui-core/src/utils/dom.js')
jest.mock('@/packages/chakra-ui-core/src/utils/generators.js')
jest.mock('v-scroll-lock', () => ({}))
jest.mock('@/packages/chakra-ui-core/src/Toast/index.js', () => {})

const renderComponent = (props) => {
  const inlineAttrs = (props && props.inlineAttrs) || ''
  const base = {
    components: { Portal, Button, AlertDialog, AlertDialogContent, AlertDialogBody, AlertDialogFooter, AlertDialogOverlay, AlertDialogHeader },
    provide: () => defaultProviders(),
    data: () => ({
      isOpen: false
    }),
    methods: {
      open () {
        this.isOpen = true
      },
      close () {
        this.isOpen = false
      }
    },
    template: `
    <div>
        <AlertDialog ${inlineAttrs}>
            <AlertDialogOverlay  />
            <AlertDialogContent data-testid="overlay">
            <AlertDialogHeader>Dialog Header</AlertDialogHeader>
            <AlertDialogBody>
            Are you sure?
              <Input data-testid="inputInsideDrawer" ref="inputInsideDrawer" placeholder="Type here..." />
            </AlertDialogBody>
            <AlertDialogFooter>
                <Button ref="cancelRef" @click="close" data-testid="close-btn">Cancel</Button>
                <Button variantColor="red" @click="close" ml="3">Delete</Button>
            </AlertDialogFooter>
            </AlertDialogContent>
        </AlertDialog>
    </div>
    `,
    ...props
  }
  return render(base, {}, vue => vue.directive('scroll-lock', jest.fn()))
}

it('should render correctly', async () => {
  useId.mockReturnValueOnce('1')
  wrapEvent.mockImplementation(() => jest.fn())
  const inlineAttrs = 'isOpen'
  const { asFragment } = renderComponent({ inlineAttrs })

  await Vue.nextTick()

  expect(asFragment(document.body.innerHTML)).toMatchSnapshot()
})

test('clicking the close button calls the onClose callback', async () => {
  const onClose = jest.fn()
  const inlineAttrs = `isOpen :on-close="close"`
  const { getByTestId } = renderComponent({ inlineAttrs, methods: { close: onClose } })

  await Vue.nextTick()
  userEvent.click(getByTestId('close-btn'))

  expect(onClose).toHaveBeenCalled()
})

test('pressing "esc" calls the onClose callback', async () => {
  const onClose = jest.fn()
  const inlineAttrs = `:isOpen="isOpen" :on-close="close"`
  const { getByTestId } = renderComponent({ inlineAttrs, data: () => ({ isOpen: true }), methods: { close: onClose } })

  await Vue.nextTick()
  const inputInside = getByTestId('inputInsideDrawer')

  fireEvent.keyDown(inputInside, { key: 'Escape' })

  expect(onClose).toHaveBeenCalled()
})

test('clicking overlay calls the onClose callback', async () => {
  const onClose = jest.fn()
  const inlineAttrs = `:isOpen="isOpen" :on-close="close"`
  const { getByTestId } = renderComponent({ inlineAttrs, data: () => ({ isOpen: true }), methods: { close: onClose } })

  await Vue.nextTick()
  const overlay = getByTestId('overlay')

  userEvent.click(overlay)

  expect(onClose).toHaveBeenCalled()
})

// TODO: A11y
// ⚠️ remove skip
it.skip('should have proper aria', async () => {
  const inlineAttrs = 'isOpen'
  const { getByTestId } = renderComponent({ inlineAttrs })

  await Vue.nextTick()
  const dialog = getByTestId('content') // eslint-disable-line

  // TODO: A11y role dialog or alertdialog?
  // expect(dialog).toHaveAttribute('role', 'dialog')
  // TODO: A11y aria modal?
  // expect(dialog).toHaveAttribute('aria-modal', 'true')
  // TODO: describedBy?
  // expect(dialog).toHaveAttribute('aria-describedby', 'chakra-dialog--body-3')
})
