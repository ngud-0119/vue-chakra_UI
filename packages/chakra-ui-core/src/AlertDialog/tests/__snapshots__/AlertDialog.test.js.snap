// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <!---->
    </div>
  </div>
  <span
    id="modal-portal-1"
  >
    <div
      class="css-0 css-79elbk"
    >
      <div
        style="position: fixed; top: 0px; right: 0px; bottom: 0px; left: 0px;"
      >
        <div
          class="css-l5w8ps"
        />
        <div
          class="css-aibg9d"
          data-testid="overlay"
          role="alertdialog"
        >
          <section
            aria-modal="true"
            class="css-i76yoi"
            id="alert-dialog-1"
            role="dialog"
            tabindex="-1"
          >
            <div
              class="css-kniazf"
              id="alert-dialog-1-label"
            >
              Dialog Header
            </div>
             
            <div
              class="css-5m10fw"
              id="alert-dialog-1-desc"
            >
              
            Are you sure?
              
              <input
                data-testid="inputInsideDrawer"
                placeholder="Type here..."
              />
            </div>
             
            <footer
              class="css-plnu2w"
            >
              <button
                class="css-0 css-rup5mm"
                data-testid="close-btn"
                tabindex="0"
                type="button"
              >
                Cancel
              </button>
               
              <button
                class="css-0 css-mbmse4"
                tabindex="0"
                type="button"
              >
                Delete
              </button>
            </footer>
          </section>
        </div>
      </div>
    </div>
  </span>
</DocumentFragment>
`;
